Default icon initialize issue

LeeVangraefschepe opened the pull request:
Initialize now gets called again in the run function of leap engine. We can do this because the default icon is already in binary format and does not need the graphics engine to get it parsed.

- Default icon is now done in window constructor
- Removed includes from Leap.h
- Title for LeapEngine now uses const char* instead of std::string
- After initialize function is added and should be used for engine settings (not setting services)

LeeVangraefschepe committed:
Created binary version of the default logo

The code is saved in the task in hacknplan and should be reimplemented later on.

LeeVangraefschepe committed:
Implemented binary loading on creation of the window

LeeVangraefschepe committed:
LeapEngine cleanup

- Removed now unused logo.png
- Removed SetIcon function in constructor
- Leap.h has no longer any includes
- Constructor is explicit and asks for const char* instead of std::string
- Renderer initialize is now done in the run function

AtlantiaKing commented on LeapEngine/Leap.cpp:
```diff
@@ -68,6 +62,9 @@ leap::LeapEngine::~LeapEngine()
 {
+    auto& renderer{ ServiceLocator::GetRenderer() };
+    renderer.Initialize();
+
```
Add function call where the user can set a custom icon

LeeVangraefschepe committed:
Added afterInitialize function

If a user wants to change settings it should be done in this function call.
You still have to set the service locators like custom rendering before this call.
